src/main/scala/com/example/Service/ReservationService.scala

```diff
@@ -0,0 +1,148 @@
+    Behaviors.setup[ReservationServiceCommand] { context =>
+      implicit val timeout: Timeout = 3.seconds
+      Behaviors.receiveMessage {
+        case CreateReservation(reservation, replyTo) =>
+          val ticketsSum = reservation.adult_tickets + reservation.student_tickets + reservation.child_tickets
+
+          if(reservation.seats.isEmpty || ticketsSum > reservation.seats.length)
+            replyTo ! Left(WrongParameter("Wrong seats"))
+          else {
+            val properName: Regex = "^[A-ZŻŹĆĄŚĘŁÓŃ][a-zżźćńółęąś]{2,}$".r
+
+            val splitNames = reservation.name match {
+              case s"$name $surname1-$surname2" => Some(List(name, surname1, surname2))
+              case s"$name $surname" => Some(List(name, surname))
+              case _ => None
+            }
+
+            splitNames match {
+              case Some(list) if list.forall(s => properName.matches(s)) =>
+                context.askWithStatus(registry, (ref: ActorRef[StatusReply[ReservationsScreeningDB]]) => GetReservationsScreeningDB(reservation.screeningId, ref)) {
+                  case Success(ReservationsScreeningDB(reservedSeats)) =>         GetReservationsScreeningDBResponse(reservedSeats, reservation, replyTo)
+                  case Failure(StatusReply.ErrorMessage(errorMessage)) =>         CreateReservationDBResponse(Left(ExceptionError(errorMessage)), replyTo)
+                  case Failure(exception) =>                                      CreateReservationDBResponse(Left(ExceptionError(exception.getMessage)), replyTo)
+                }
+              case _ =>                                                           replyTo ! Left(WrongParameter("Wrong name"))
+            }
+          }
+          Behaviors.same
+
+        case GetReservationsScreeningDBResponse(reservedSeats, reservation, replyTo) =>
+            context.askWithStatus(registry, (ref: ActorRef[StatusReply[Option[ScreeningScreeningRoomDB]]]) => GetScreeningScreeningRoomDB(reservation.screeningId, ref)) {
+              case Success(Some(screeningRoom)) =>
+                if(!reservation.seats.exists( s =>
+                    reservedSeats.contains(s) ||
+                    s._1 < 1 ||
+                    s._2 < 1 ||
+                    s._1 > screeningRoom.rows ||
+                    s._2 > screeningRoom.seatsPerRow))
+                {
+                  val seats = for {
+                    r <- 1 to screeningRoom.rows
+                    s <- 1 to screeningRoom.seatsPerRow
+                  } yield (r,s)
+
+                  val availableSeatsMap = seats.groupMap(_._1)(seat => if(!reservation.seats.contains(seat) && !reservedSeats.contains(seat)) false else true)
+                  val listOfSeatsByThree = availableSeatsMap.values.flatMap(row => row.iterator.sliding(3).withPartial(false).toList)
+                  if(listOfSeatsByThree.exists(_ == List(true, false, true)))
```

Comment:
This validation logic seems really complex, why not extract it to some other class?

Reply:
it would be also easier to test it